voluseg/_steps/step1.py

```diff
@@ -84,34 +103,54 @@ def make_output_volume(name_volume, volume):
                 # pad planes as necessary
                 if p.registration and p.planes_pad:
                     volume = np.lib.pad(
-                        volume, ((0, 0), (0, 0), (p.planes_pad, p.planes_pad)),
-                        'constant', constant_values=(np.percentile(volume, 1),)
+                        volume,
+                        ((0, 0), (0, 0), (p.planes_pad, p.planes_pad)),
+                        "constant",
+                        constant_values=(np.percentile(volume, 1),),
                     )
 
                 # save volume in output directory
                 if p.registration:
-                    save_volume(fullname_volume+ori+nii, volume, p.affine_mat)
+                    save_volume(fullname_volume + ori + nii, volume, p.affine_mat)
                 else:
                     volume = volume.T
-                    save_volume(fullname_volume+ali+hdf, volume)
+                    save_volume(fullname_volume + ali + hdf, volume)
+
             # end make output volume
 
             # get full name of input volume, input data and list of planes
-            fullname_volume_input = tuple_fullname_volume_input[1]
-            volume = load_volume(fullname_volume_input+p.ext)
-            if len(p.input_dirs) == 1:
-                dir_prefix = None
+            if parameters.get("ext") == ".nwb":
```

Comment:
This code seems to convert a 4D volume to a sequence of 3D volumes in all cases. However, I think it may be more elegant to process 4D volumes natively where possible. This would be especially the case when the images are already aligned, and so can be directly segmented without needing to invoke ANTs (i.e. bypass step2).

Reply:
@mikarubi Indeed, all I did here was to ingest data coming from a nwb file in the same way as other data is processed.
Is voluseg capable of processing 4D structures in another way?

Reply:
It's one of the proposed changes we have in the scope of work. We should be able to make this work without too much trouble if we add the relevant functionality to these functions:

* https://github.com/mikarubi/voluseg/blob/master/voluseg/_tools/save_volume.py
* https://github.com/mikarubi/voluseg/blob/master/voluseg/_tools/load_volume.py

(i.e. if detect 4D file then load/save a part of that file)

I talked about this sometime with a colleague and can loop him in in case he has some ideas about how to best implement this.